cvmfs/catalog_mgr_rw.cc

```diff
@@ -1253,6 +1286,11 @@ void WritableCatalogManager::CatalogUploadCallback(
   assert(catalog_size > 0);
 
   SyncLock();
+
+  if (useLocalCache()) {
+    CopyCatalogToLocalCache(result);
+  }
```

Comment:
Copying can happen outside the lock.
```suggestion
  if (useLocalCache()) {
    CopyCatalogToLocalCache(result);
  }
  
  SyncLock();
```